arangod/RestHandler/RestAdminClusterHandler.cpp

```diff
@@ -0,0 +1,1563 @@
+              VPackBuilder builder(payload);
+              VPackObjectBuilder ob(&builder);
+
+              // append all the job keys
+              builder.add(VPackObjectIterator(job));
+              builder.add("error", VPackValue(false));
+              builder.add("job", VPackValue(jobId));
+              builder.add("status", VPackValue(path[2]));
+            }
+
+            resetResponse(rest::ResponseCode::OK);
+            response()->setPayload(std::move(payload), true);
+            return;
+          }
+        }
+
+        generateError(rest::ResponseCode::NOT_FOUND, TRI_ERROR_HTTP_NOT_FOUND);
+      } else {
+        generateError(result.asResult());
+      }
+    }).thenError<VPackException>([this](VPackException const& e) {
+      generateError(Result{e.errorCode(), e.what()});
+    }).thenError<std::exception>([this](std::exception const& e) {
+      generateError(rest::ResponseCode::SERVER_ERROR, TRI_ERROR_HTTP_SERVER_ERROR, e.what());
+    }));
+}
+
+RestStatus RestAdminClusterHandler::handleSingleServerJob(std::string const& job) {
+
+  if (!ExecContext::current().isAdminUser()) {
+    generateError(rest::ResponseCode::FORBIDDEN, TRI_ERROR_HTTP_FORBIDDEN);
+    return RestStatus::DONE;
+  }
+
+  if (!ServerState::instance()->isCoordinator()) {
+    generateError(rest::ResponseCode::FORBIDDEN, TRI_ERROR_HTTP_FORBIDDEN, "only allowed on coordinators");
+    return RestStatus::DONE;
+  }
+
+  if (request()->requestType() != rest::RequestType::POST) {
+    generateError(rest::ResponseCode::METHOD_NOT_ALLOWED, TRI_ERROR_HTTP_METHOD_NOT_ALLOWED);
+    return RestStatus::DONE;
+  }
+
+  bool parseSuccess;
+  VPackSlice body = parseVPackBody(parseSuccess);
+  if (!parseSuccess) {
+    return RestStatus::DONE;
+  }
+
+  if (body.isObject()) {
+    VPackSlice server = body.get("server");
+    if (server.isString()) {
+      std::string serverId = resolveServerNameID(server);
+      return handleCreateSingleServerJob(job, serverId);
+    }
+  }
+
+  generateError(rest::ResponseCode::BAD, TRI_ERROR_BAD_PARAMETER, "object with key `server`");
+  return RestStatus::DONE;
+}
+
+RestStatus RestAdminClusterHandler::handleCreateSingleServerJob(std::string const& job, std::string const& serverId) {
+  std::string jobId = std::to_string(server().getFeature<ClusterFeature>().clusterInfo().uniqid());
+  auto jobToDoPath = arangodb::cluster::paths::root()->arango()->target()->toDo()->job(jobId);
+
+  VPackBuilder builder;
+  {
+    VPackObjectBuilder ob(&builder);
+    builder.add("type", VPackValue(job));
+    builder.add("server", VPackValue(serverId));
+    builder.add("jobId", VPackValue(jobId));
+    builder.add("creator", VPackValue(ServerState::instance()->getId()));
+    builder.add("timeCreated", VPackValue(timepointToString(std::chrono::system_clock::now())));
+  }
+
+  return waitForFuture(AsyncAgencyComm().setValue(20s, jobToDoPath, builder.slice())
+    .thenValue([this, jobId = std::move(jobId)](AsyncAgencyCommResult &&result) {
```

Comment:
```suggestion
    .thenValue([this, jobId = std::move(jobId)](AsyncAgencyCommResult&& result) {
```
do we need to capture `self` here?

Reply:
This lambda is executed (or destroyed) strictly before the `thenFinal` lambda in `waitForFuture` which captures `self`.